### ISSUE TITLE
MA0158 is not reported when lock is initialized in .ctor or .cctor

### ISSUE THREAD

**Issue by drieseng** [meziantou/Meziantou.Analyzer#1032]
### Version of the Meziantou.Analyzer NuGet package

3.0.11

### Rule Identifier

MA0158

### Target Framework

.NET 10.0

### C# Language version

C# 14

### Description

When the lock field is initialized in the instance or static constructor, MA0158 is not reported if the type of the field is not **System.Threading.Lock**.

### Reproduction Steps

**Constructor**:
```c#
public sealed class A
{
    private readonly object _lock; // MA0158 is not reported

    public A()
    {
        _lock = new object();
    }

    public void Run()
    {
        lock (_lock)
        {
        }
    }
}
```

**Type initializer / .cctor**:
```c#
public sealed class B
{
    private static readonly object Lock; // MA0158 is not reported

    static B()
    {
        Lock = new object();
    }

    public void Run()
    {
        lock (Lock)
        {
        }
    }
}
```
